az-digital/az_quickstart
# Closes #807, Closes #1987, #2446 Switch to CKEditor5 and update text editor and format settings.

## Description

Alternate CKEditor5 PR that _does not_ move `az_barrio`'s Arizona Bootstrap theme settings to `az_core` like #2596 does.

- Upgrades `az_standard` text format and editor configs to CKEditor5
- Upgrades `basic` text format and editor configs to CKEditor5
- Installs CKEditor5 module instead of CKEditor (4) module by default on new site installs
- Adds DB update to install CKEditor5 module on existing sites
- Updates icons for custom embed buttons to SVGs (needed for CKEditor5)
- Adds CKEditor5 stylesheets to AZ Barrio theme
- Moves most of `az_barrio`'s `hook_library_info_alter()` implementation (and associated helper functions) to `az_core` so AZ Barrio libraries (including Arizona Bootstrap) are loaded correctly in admin theme

Questions we still need to consider:
- Is it going to be safe for people to apply these config changes via Config Distro if they have customized the text formats or editors in any way?
- When (if ever) do we want to automate the uninstallation of the contrib ckeditor module?
- Should we remove the remaining CKeditor 4 specific things from Quickstart now or later? (AZ Barrio ckeditor styles, `az_media_ckeditor_styles_alter()`, etc)

See validation messages that came up when switching editor/text_format configs to CKeditor5 here: https://github.com/az-digital/az_quickstart/issues/1987#issuecomment-1640724617

## Related issues





- Closes #807
- Closes #1987
- #2446

## How to test
https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2769.probo.build/
Verify that editor works with new and existing content at least as well as it did with CKEditor 4.


## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
